API.md

```diff
@@ -652,7 +652,7 @@ For example, use `is: Joi.number().required()` to guarantee that a joi reference
 
 ```js
 const schema = {
-    a: Joi.any().valid('x').when('b', { is: 5, then: Joi.valid('y'), otherwise: Joi.valid('z') }),
```

Comment:
I wanted to use `when` for an example with `exist`. I think the `in` property of a `when` definition is a particular place where `exist` makes more sense than `required` and is good to highlight with an example.

The reason that I replaced the `5` is that the first 2 examples of `when` both have static values for the `is` property. I thought adding more variation to the existing examples would be preferable to adding a 5th example specifically to show this. 

If you would prefer the `exist` example be moved or changed, I'm happy to do that.